src/openloco/windows/LandscapeGeneration.cpp

```diff
@@ -0,0 +1,1371 @@
+                options::widgets,
+                land::widgets,
+                forests::widgets,
+                towns::widgets,
+                industries::widgets,
+            };
+
+            widget_t* newWidgets = widgetCollectionsByTabId[window->current_tab];
+            if (window->widgets != newWidgets)
+            {
+                window->widgets = newWidgets;
+                window->init_scroll_widgets();
+            }
+
+            static const widx tabWidgetIdxByTabId[] = {
+                tab_options,
+                tab_land,
+                tab_forests,
+                tab_towns,
+                tab_industries,
+            };
+
+            window->activated_widgets &= ~((1 << tab_options) | (1 << tab_land) | (1 << tab_forests) | (1 << tab_towns) | (1 << tab_industries));
+            window->activated_widgets |= (1ULL << tabWidgetIdxByTabId[window->current_tab]);
+        }
+
+        // 0x004CD320
+        static void sub_4CD320(window* window)
+        {
+            registers regs;
+            regs.esi = (int32_t)window;
+
+            call(0x004CD320, regs);
+        }
+
+        // 0x0043DC98
+        static void switchTab(window* window, widget_index widgetIndex)
+        {
+            if (input::is_tool_active(window->type, window->number))
+                input::cancel_tool();
+
+            window->current_tab = widgetIndex - widx::tab_options;
+            window->frame_no = 0;
+            window->flags &= ~(window_flags::flag_16);
+            window->disabled_widgets = 0;
+
+            static const uint64_t* enabledWidgetsByTab[] = {
+                &options::enabled_widgets,
+                &land::enabled_widgets,
+                &forests::enabled_widgets,
+                &towns::enabled_widgets,
+                &industries::enabled_widgets,
+            };
+
+            window->enabled_widgets = *enabledWidgetsByTab[window->current_tab];
+
+            static const uint64_t* holdableWidgetsByTab[] = {
+                &options::holdable_widgets,
+                &land::holdable_widgets,
+                &forests::holdable_widgets,
+                &towns::holdable_widgets,
+                &industries::holdable_widgets,
+            };
+
+            window->holdable_widgets = *holdableWidgetsByTab[window->current_tab];
+
+            static window_event_list* eventsByTab[] = {
+                &options::events,
+                &land::events,
+                &forests::events,
+                &towns::events,
+                &industries::events,
+            };
+
+            window->event_handlers = eventsByTab[window->current_tab];
+
+            switchTabWidgets(window);
+
+            window->invalidate();
+
+            const gfx::ui_size_t* newSize = &window_size;
+            if (widgetIndex == widx::tab_land)
+                newSize = &land_tab_size;
+
+            window->min_width = window->max_width = window->width = newSize->width;
```

Comment:
`window->set_size(*newSize);`

Reply:
Ah, cheers. I did check whether we had a function for it in window.cpp, but I hadn't considered it might be header-only!